src/plugins/intel_cpu/src/memory_state.cpp

```diff
@@ -297,18 +297,19 @@ void VariableStateKVcache::set_state_impl(const ov::SoPtr<ov::ITensor>& state) {
         auto S = internal.size(3);
         auto nthr = parallel_get_max_threads();
         std::vector<PlainTensor> buffers(nthr);
+        m_scale_zp.resize<float>({L0, B, H, 2});
```

Comment:
Could you please clarify why we need these changes? Is there smt we need to cover with tests?

Reply:
The current logic is not correct if kvcache type is u8, because the memory of scale and zero point is not changed for the new state shape. The case [ConcatSDPTransposeTestSetState](https://github.com/openvinotoolkit/openvino/blob/81d34114b218633a9fd4f670077870dc0da7225f/src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/concat_transpose_sdp_transpose.cpp#L408C8-L408C38) could cover now after setting the default u8 kv cache.

Reply:
So it means U8 cache on current master is broken? 
Can we extend ConcatSDPTransposeTestSetState with additional parameter that will verify different KV cache precision: {fp32, fp16, bf16, u8}?

Reply:
Updated. Although `f16` is not covered but actually there are 2 paths in [`set_state`](https://github.com/openvinotoolkit/openvino/blob/0206cd12601687cbdaa96e45f7753f388a9b25c0/src/plugins/intel_cpu/src/memory_state.cpp#L287), so `u8` and non-u8 should be enough to cover all the paths in `set_state`.